File: Java/Sorpresa_IrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ilzk
 */
public class Sorpresa_IrCarcel extends Sorpresa {
    
    /**Constructor de sorpresa que crea la sorpresa que envia a la carcel
     * 
     * @param _tipo el tipo de la sorpresa a crear
     * @param _tablero el tablero del que se obtiene el numCasillaCarcel
     */
    
    public Sorpresa_IrCarcel(Tablero _tablero) {
            texto = "El jugador va a la carcel";
            tablero = _tablero;
            valor = tablero.getCarcel();
    }
    
    @Override
    /**Aplica la sorpresa Ir a carcel al jugador actual
     * 
     */
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual,todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    
    /**Sobreescribe el metodo toString
     * 
     */
    public String toString(){
        return "Ir carcel";
    }
}

File: Ruby/sorpresa_ir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'sorpresa'
module Civitas
  class Sorpresa_ir_carcel < Sorpresa
    def initialize(_tablero)
      #Sorpresa.new("Ir carcel", -1, nil, _tablero)
      @texto = "Ir carcel"
      @tablero = _tablero
    end

    def aplicar_a_jugador(actual, todos)
      if (jugador_correcto(actual, todos))
          informe(actual, todos)
          todos[actual].encarcelar(@tablero.num_casilla_carcel)
      end
    end

    def to_s()
      "Ir carcel"
    end

    public_class_method:new
  end
end